shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -964,6 +964,57 @@ - (void)keyboardWillBeHidden:(NSNotification*)notification {
   [self updateViewportMetrics];
 }
 
+- (void)dispatchPresses:(NSSet<UIPress*>*)presses API_AVAILABLE(ios(13.4)) {
+  if (@available(iOS 13.4, *)) {
+    for (UIPress* press in presses) {
+      if (press.key == nil) {
+        continue;
+      }
+      NSMutableDictionary* keyMessage = [@{
+        @"keymap" : @"ios",
+        @"type" : @"unknown",
+        @"keyCode" : @(press.key.keyCode),
+        @"modifiers" : @(press.key.modifierFlags),
+        @"characters" : press.key.characters,
+        @"charactersIgnoringModifiers" : press.key.charactersIgnoringModifiers
+      } mutableCopy];
+
+      if (press.phase == UIPressPhaseBegan) {
+        keyMessage[@"type"] = @"keydown";
+      } else if (press.phase == UIPressPhaseEnded || press.phase == UIPressPhaseCancelled) {
+        keyMessage[@"type"] = @"keyup";
+      }
```

Comment:
So does this mean `pressesChanged` events will have a "unknown" type? Does the framework process this at all?

Reply:
Good point. The framework throws an error in this case. It probably would be more sensible to just ignore phases which are not processed by the framework:
- UIPressPhaseStationary ("A button was pressed but hasn’t moved since the previous event.") probably is not very useful and could be skipped
- UIPressPhaseChanged ("A button moved, or the force property changed.") i don't think that any physical keyboard for iOS has a button with different force or directional values. I assume that this is for gamepads or other types of inputs.

Reply:
Yeah. I think we can ignore this case until we're sure we want the framework to throw an error or have added an implementation.